FreeRTOS_IP.c

```diff
@@ -3889,53 +3868,157 @@ uint32_t FreeRTOS_GetIPAddress( void )
 }
 /*-----------------------------------------------------------*/
 
-#if 0
+#if ( ipconfigCOMPATIBLE_WITH_SINGLE != 0 )
 
 /*
- * The helper functions here below can not exist in a multi-interface environment.
+ * The helper functions here below assume that there is a single
+ * interface and a single end-point (ipconfigCOMPATIBLE_WITH_SINGLE)
+ */
+
+/**
+ * @brief Set the IP-address of device.
+ *
+ * @param ulIPAddress: The new IP-address.
  */
     void FreeRTOS_SetIPAddress( uint32_t ulIPAddress )
     {
         /* Sets the IP address of the NIC. */
-        *ipLOCAL_IP_ADDRESS_POINTER = ulIPAddress;
+        NetworkEndPoint_t * pxEndPoint = FreeRTOS_FirstEndPoint( NULL );
+
+        if( pxEndPoint != NULL )
+        {
+            pxEndPoint->ipv4_settings.ulIPAddress = ulIPAddress;
+        }
     }
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Get the gateway address of the subnet.
+ *
+ * @return The IP-address of the gateway, zero if a gateway is
+ *         not used/defined.
+ */
     uint32_t FreeRTOS_GetGatewayAddress( void )
     {
-        return xNetworkAddressing.ulGatewayAddress;
+        uint32_t ulIPAddress = 0U;
+        NetworkEndPoint_t * pxEndPoint = FreeRTOS_FirstEndPoint( NULL );
+
+        if( pxEndPoint != NULL )
+        {
+            ulIPAddress = pxEndPoint->ipv4_settings.ulGatewayAddress;
+        }
+
+        return ulIPAddress;
     }
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Get the DNS server address.
+ *
+ * @return The IP address of the DNS server.
+ */
     uint32_t FreeRTOS_GetDNSServerAddress( void )
     {
-        return xNetworkAddressing.ulDNSServerAddress;
+        uint32_t ulIPAddress = 0U;
+        NetworkEndPoint_t * pxEndPoint = FreeRTOS_FirstEndPoint( NULL );
```

Comment:
Same comment as the one regarding gateway address.

Reply:
The same answer as to `FreeRTOS_GetGatewayAddress()` : the function is only defined when `ipconfigCOMPATIBLE_WITH_SINGLE` is used.